examples/architectureComponentsExample/src/main/java/io/realm/examples/arch/livemodel/LiveRealmObject.java

```diff
@@ -26,27 +27,44 @@
 
 
 public class LiveRealmObject<T extends RealmModel> extends LiveData<T> {
+    // The listener will listen until the object is deleted.
+    // An invalidated object shouldn't be set in LiveData, null is set instead.
     private RealmObjectChangeListener<T> listener = new RealmObjectChangeListener<T>() {
         @Override
-        public void onChange(T object, ObjectChangeSet objectChangeSet) {
-            LiveRealmObject.this.postValue(object);
+        public void onChange(@NonNull T object, ObjectChangeSet objectChangeSet) {
+            if (!objectChangeSet.isDeleted()) {
+                setValue(object);
+            } else {
+                setValue(null);
+            }
         }
     };
 
     @MainThread
-    public LiveRealmObject(T object) {
+    public LiveRealmObject(@NonNull T object) {
+        //noinspection ConstantConditions
+        if (object == null) {
+            throw new IllegalArgumentException("The object cannot be null!");
+        }
         setValue(object);
     }
 
+    // We should start observing and stop observing, depending on whether we have observers.
     @Override
     protected void onActive() {
         super.onActive();
-        RealmObject.addChangeListener(getValue(), listener);
+        T object = getValue();
+        if (object != null && RealmObject.isValid(object)) {
+            RealmObject.addChangeListener(object, listener);
+        }
     }
 
     @Override
     protected void onInactive() {
         super.onInactive();
-        RealmObject.removeChangeListener(getValue(), listener);
+        T object = getValue();
+        if (object != null && RealmObject.isValid(object)) {
```

Comment:
`onInactive` is bound to observing the LiveData, and can contain null if the object was deleted. Also, object could have its Realm instance invalidated if we're unlucky.

A listener should not be removed from an invalid object, because invalid objects throw when accessed even if you try to remove listeners from them, if I recall correctly.

The intent here is that this can only be null if the object was deleted, which mimics the behavior of `findFirst().addChangeListener` (calls with `isDeleted == true` change set once more, and then never again).

Now that you mention it, I should probably add the valid check to the RealmResults' live data too, its Realm instance can be closed, after all.